components/Chat/Chat.vue

```diff
@@ -751,9 +751,21 @@
         voice.rate   = this.getTtsRate / 10.0;
         voice.volume = this.getTtsVolume / 10.0;
         voice.pitch  = pitch;
-        // Completely sanitized messages (i.e. emotes only) don't get read
-        // TODO: implement empty message filter
-        voice.text   = ( this.getTtsReadUsername ? `${username} says: ` : '' ) + message;
+        voice.text   = message;
+
+        if( this.getTtsReadUsername ) {
+
+          // Since 'username says' will be prepended to the read message, we have to ensure
+          // that the message isn't blank; skip only if there isn't readable text.
+          const pattern = new RegExp( '\S', 'gi' );
```

Comment:
`new RegExp( '\S', 'gi' );` ends up being `/S/gi` 
should be 
`new RegExp( /\S/, 'gi' );` which computes to `/\S/gi`.
(or you need to escape the backslash like: `'\\S', 'gi'`)